File: CSharp.Functions/Common/DomainServices/RestServices/AuthService/IAuth0Service.cs
using Common.Entities;

namespace Common
{
    public interface IAuth0Service
    {
        Task<AccessToken> GetAccessTokenForSignUpAsync();
        Task<Auth0User?> CreateOnboardingUserAsync(OnboardingUserBase onboardingUser);
        Task<bool> DeleteUserAsync(string userId);
        Task<HttpResponseMessage> ChangePasswordAsync(string email, string newPassword);
        void Register(string clientId, string clientSecret, string baseUrl, string apiToken);
    }
}

File: vue.jsonforms/src/infra/dependency-services/rest/auth0/IAuth0Service.ts
export declare interface IAuth0Service {
  AuthenticateAsync(accessToken: string, idToken: string): Promise<string>;

  getAccessTokenForSignUpAsync(): Promise<{
    access_token: string;
    id_token: string;
  }>;

  getTokenForSignupAndSaveLocallyAsync(): Promise<boolean>;

  changePasswordAsync(
    email: string,
    newPassword: string,
    onboardingType: string,
  ): Promise<boolean>;
}

export const IAuth0ServiceInfo = {
  name: "IAuth0Service",
};